public/app/features/variables/editor/reducer.ts

```diff
@@ -65,6 +82,7 @@ const variableEditorReducerSlice = createSlice({
       state: VariableEditorState,
       action: PayloadAction<{ propName: string; propValue: any }>
     ) => {
+      // @ts-ignore - temp ignoring the errors now the state type is more strict
```

Comment:
Hmm, is this necessary?

Reply:
It is, because `extended` cannot be an empty object (with superfluous properties).

`state.extended` is nullable, and spreading null into an object gives you an empty object. Typescript then complains that we aren't setting `extended` to `AdHocVariableEditorState | DataSourceVariableEditorState | QueryVariableEditorState`, which we aren't because `propName` is a string (so we/ts don't know what it is).

This is how this state is currently untyped, so I don't think this regresses anything. It just ignores an error we weren't (but should have been) catching before.

Fixing this is subject of the [whole separate PR](https://github.com/grafana/grafana/pull/44957/files#diff-4a995a538d05b0805440eca25f743a6521027148fec55afcd1ba60e57a97c40aL85) which should all be ready to go in at the same time, so I'm hoping this will fly just for a few minutes 😇

Reply:
Thanks for the explanation and I've approved the PR anyway 👍